cliconfig/config.go

```diff
@@ -117,49 +186,19 @@ func Load(configDir string) (*ConfigFile, error) {
 	if _, err := os.Stat(confFile); err != nil {
 		return &configFile, nil //missing file is not an error
 	}
-
-	b, err := ioutil.ReadFile(confFile)
+	file, err := os.Open(confFile)
 	if err != nil {
 		return &configFile, err
 	}
-
-	if err := json.Unmarshal(b, &configFile.AuthConfigs); err != nil {
-		arr := strings.Split(string(b), "\n")
-		if len(arr) < 2 {
-			return &configFile, fmt.Errorf("The Auth config file is empty")
-		}
-		authConfig := AuthConfig{}
-		origAuth := strings.Split(arr[0], " = ")
-		if len(origAuth) != 2 {
-			return &configFile, fmt.Errorf("Invalid Auth config file")
-		}
-		authConfig.Username, authConfig.Password, err = DecodeAuth(origAuth[1])
-		if err != nil {
-			return &configFile, err
-		}
-		origEmail := strings.Split(arr[1], " = ")
-		if len(origEmail) != 2 {
-			return &configFile, fmt.Errorf("Invalid Auth config file")
-		}
-		authConfig.Email = origEmail[1]
-		authConfig.ServerAddress = defaultIndexserver
-		configFile.AuthConfigs[defaultIndexserver] = authConfig
-	} else {
-		for k, authConfig := range configFile.AuthConfigs {
-			authConfig.Username, authConfig.Password, err = DecodeAuth(authConfig.Auth)
-			if err != nil {
-				return &configFile, err
-			}
-			authConfig.Auth = ""
-			authConfig.ServerAddress = k
-			configFile.AuthConfigs[k] = authConfig
-		}
-	}
-	return &configFile, nil
+	err = configFile.LegacyLoadFromReader(file)
+	return &configFile, err
 }
 
 // Save encodes and writes out all the authorization information
 func (configFile *ConfigFile) Save() error {
+	if configFile.Filename() == "" {
```

Comment:
I would imaging generating an error should be expected behavior. It should be worse to save, get no error, and then expect there to be a file when there is none. While an empty filename may not be an invalid state, trying to save an empty filename should be an invalid action.

I added the function to save to a writer though.